### PR TITLE
Update name transformer

### PR DESCRIPTION
**Pull request by ArcturusZhang**
# Description

In the previous implementation, we did not transform the name of choice values. This PR also transforms that.

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs [hunk 1]
```diff
@@ -86,9 +88,24 @@ private static void TransformSchema(Schema schema, NameTransformer transformer,
             }
         }
 
-        private static void TransformBasicSchema(Schema schema, NameTransformer transformer, ConcurrentDictionary<string, string> wordCache)
+        private static void TransformChoiceSchema(ChoiceSchema choiceSchema, NameTransformer transformer, ConcurrentDictionary<string, string> wordCache)
```
Comment (AlexanderSher) [inline src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs:91]: What's the difference between `TransformChoiceSchema` and `TransformSealedChoiceSchema`? They look identical to me.

Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs:91]: Yes, they are identical. But unfortunately `ChoiceSchema` and `SealedChoiceSchema` does not inherit from each other, they just share the same parent, and the most important property `Choices` is individually defined in these two classes.
I do not want to write too many lines in the switch statement, therefore I create these two functions to do all the things.

Comment (AlexanderSher) [reply src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs:91]: Will this work:

```cs
private static void TransformChoiceSchema(Languages languages, ICollection<ChoiceValue> choices, NameTransformer transformer, ConcurrentDictionary<string, string> wordCache) 
{
    TransformLanguage(languages, transformer, wordCache);
    TransformChoices(choices, transformer, wordCache);
}
```

```cs
case ChoiceSchema choiceSchema:
    TransformChoiceSchema(choiceSchema.Language, choiceSchema.Choices, transformer, wordCache);
    break;
case SealedChoiceSchema sealedChoiceSchema:
    TransformChoiceSchema(sealedChoiceSchema.Language, sealedChoiceSchema.Choices, transformer, wordCache);
    break;
```

**Review by Yao725 — APPROVED** [review]

File: src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs (hunk 1 above)
Comment (ArcturusZhang) [reply src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs:91]: Let me do this change

**Pull request merged**